client/blocks/reader-post-card/featured-video.jsx

```diff
@@ -15,7 +15,7 @@ import FeaturedImage from './featured-image';
 class FeaturedVideo extends React.Component {
 
 	static propTypes = {
-		thumbnailUrl: React.PropTypes.string,
+		thumbnailUrlPromise: React.PropTypes.object,
```

Comment:
So looking further I think we need 2 items to make a request, the videoId and the service. So we could pass both as a prop from the lib and create a redux action:
```
getVideoThumbnail( videoService, videoId );
```
Which we could then render a [query component](https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md#query-components) to handle the fetching boiler plate for us.

The action itself might look like:
```javascript
getVideoThumbnail( service, videoId ) {
  if ( service === 'youtube' ) {
          const thumbnailUrl = videoId ? `https://img.youtube.com/vi/${ videoId }/mqdefault.jpg` : null;
          return {
                type: VIDEO_THUMBNAIL_RECIEVE,
                thumbnailUrl
          };
   }
   if ( service === 'vimeo' ) {
        const fetchUrl = `http://vimeo.com/api/v2/video/${ videoId }.json`;
        return dispatch => {
           dispatch( { type: VIDEO_VIMEO_THUMBNAIL_REQUEST } );

           return fetch( fetchUrl, { method: 'get' } )
            .then( resp => resp.json() )
            .then( resp => {
              dispatch( {
                type: VIDEO_THUMBNAIL_RECIEVE,
                thumbnailUrl: resp[ 0 ].thumbnail_large ,
              } );
              dispatch( {
                type: VIDEO_VIMEO_THUMBNAIL_REQUEST_SUCCESS
              } );
              
            } ) )
            .catch( err => Promise.reject( err ) );
        }
   }
   return {
      type: 'VIDEO_THUMBNAIL_UNKNOWN_SERVICE'
   };
}
```

In addition to this we'll want to create a reducer to store these thumbnails, and a selector to fetch them from redux. See https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md for more background on our current data patterns

cc @blowery or @fraying let me know if you think this is out of scope